src/client/renderer/r_types.h

```diff
@@ -841,6 +877,7 @@ typedef struct {
 
 	const r_entity_t *current_entity; // entity being rendered
 	const r_shadow_t *current_shadow; // shadow being rendered
+	vec4_t current_shadow_light, current_shadow_plane;
```

Comment:
Those were added to resolve the todo in r_mesh_shadow, they store the state of the "light" and "plane" uniforms essentially. I'll see if I can find a way to keep these contained in the shadow program though.